.github/workflows/pull_request_checks.yml

```diff
@@ -10,31 +10,36 @@ on:
 jobs:
   tests:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        py-ver: ["py39", "py310", "py311", "py312"]
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        with:
-          python-version: "3.x"
-      - name: Get week number
-        id: week
-        run: echo "week=$(date '+W%V')" >> $GITHUB_OUTPUT
-      - uses: actions/cache@v3
+      - name: Cache conda environment
+        id: conda-env-cache
+        uses: actions/cache@v3
         with:
-          key: tests|${{ runner.os }}-${{ runner.arch}}|${{ steps.week.outputs.week }}|${{ hashFiles('requirements/locks/*') }}
-          path: |
-            .tox
-            ~/.cache/pip
-      - name: Install dependencies
-        run: python3 -m pip install tox tox-conda
+          key: conda|${{runner.os}}-${{runner.arch}}|${{ hashFiles(format('requirements/locks/{0}-lock-linux-64.txt', matrix.py-ver)) }}
+          path: ~/conda-env
+      - name: Create conda environment
+        if: steps.conda-env-cache.outputs.cache-hit != 'true'
+        run: |
+          # Check cache hasn't pulled a partial key match.
+          test ! -e "${HOME}/conda-env"
```

Comment:
For the conda-env, default will be $HOME but worth having this user overridable (considering conda install size, etc.)?

Reply:
Jorge can comment more generally but how portable is $HOME also, for other system architectures? Realise this env variable could be exported manually

Reply:
$HOME is part of POSIX, which is a standard definition that all (post 1990s) Unix follow, and has been a standard part of unix-like operating systems since the 1970s, so it should be pretty portable.

If the location does need to be changed, it can be updated by just replacing any reference in this file to `${HOME}/conda-env` and `~/conda-env` with the new location. I've chosen $HOME as its one of the few places I can rely on having a consistent location, which is needed for the caching to work. $TMPDIR was the other consideration, however on GitHub Actions each job gets a unique temporary directory, so the restored cache wouldn't be found.